src/components/plots/barPlot.ts

```diff
@@ -16,6 +19,9 @@ export module Plots {
     private _labelFormatter: Formatter = Formatters.identity();
     private _labelsEnabled = false;
     private _hideBarsIfAnyAreTooWide = true;
+    private _labelAreas: Utils.Map<Dataset, D3.Selection>;
+    private _labelMeasurers: Utils.Map<Dataset, SVGTypewriter.Measurers.Measurer>;
+    private _labelWriters: Utils.Map<Dataset, SVGTypewriter.Writers.Writer>;
```

Comment:
It seems like there would be less trouble if you had a single `Map` to a composite object.